From 0c5acb98920a77724ddf3c5882b0d299b5a627de Mon Sep 17 00:00:00 2001
Message-Id: <0c5acb98920a77724ddf3c5882b0d299b5a627de.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Nabendu Maiti <nabendu.bikash.maiti@intel.com>
Date: Wed, 28 Jan 2015 08:54:01 +0530
Subject: [PATCH 15/28] MUST_REBASE [VPG]: adf/intel/vlv: attach intel planes
 to the respective pipe

This patch adds code to attach planes to corresponding pipes.Doing so,
any of these elements(pipe, plane port) can be accessed from each other,
for the encoder usages.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5587
Change-Id: Ic405628192f9021ee224299a0d5b76ffad3fad53
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Nabendu Maiti <nabendu.bikash.maiti@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c |   38 ++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index 6ef37fa..108bfac 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -192,6 +192,42 @@ void vlv_dc_config_destroy(struct intel_dc_config *config)
 	return;
 }
 
+static void vlv_attach_intel_pipeplane(struct vlv_pipeline *disp, u8
+					   plane_per_pipe)
+{
+	struct intel_plane *intel_plane = NULL;
+	struct intel_pipe *intel_pipe = NULL;
+	int i = 0;
+
+	/*
+	 * In this platform the plane and pipe are fixed and cannot be
+	 * moved across to a different pipe, hence set the attachment by
+	 * default over here
+	 */
+	if (disp->type == INTEL_PIPE_DSI)
+		intel_pipe = &disp->gen.dsi.base;
+	else if (disp->type == INTEL_PIPE_DP || disp->type == INTEL_PIPE_EDP)
+		intel_pipe = &disp->gen.dp.base;
+	else if (disp->type == INTEL_PIPE_HDMI)
+		intel_pipe = &disp->gen.hdmi.base;
+
+	if (intel_pipe != NULL) {
+		/* Attach Primary plane to intel Pipe */
+		intel_plane = &disp->pplane.base;
+		if (intel_plane->ops->attach)
+			intel_plane->ops->attach(intel_plane, intel_pipe);
+
+		/* Attach Sprite[n] to the intel pipe */
+		for (i = 0; i < (plane_per_pipe - 1); i++) {
+			intel_plane = &disp->splane[i].base;
+			if (intel_plane->ops->attach)
+				intel_plane->ops->attach(intel_plane,
+							 intel_pipe);
+		}
+	}
+	return;
+}
+
 static int vlv_display_encoder_init(struct vlv_dc_config *vlv_config, int pipe,
 				int port, u8 disp_no)
 {
@@ -394,6 +430,8 @@ static int vlv_initialize_disp(struct vlv_dc_config *vlv_config,
 	vlv_pipe = &disp->pipe;
 	err = vlv_pipe_init(vlv_pipe, (enum pipe) pipe);
 
+	vlv_attach_intel_pipeplane(disp, CHV_MAX_PLANES);
+
 	/* FIXME: update from attachment */
 	pll = &disp->pll;
 	err = vlv_pll_init(pll, type, (enum pipe) pipe, port);
-- 
1.7.9.5

